transport/transport.go

```diff
@@ -306,3 +319,71 @@ func NewMultiTransport(backends []http.RoundTripper,
 		Backends:        backends,
 		HandleResponses: responsesHandler}
 }
+
+// SetTransportsConfig assign transport config to Matcher
+func (m *Matcher) SetTransportsConfig(clientConfig httphandlerConfig.Client) {
+	m.TransportsConfig = clientConfig.Transports
+}
+
+// SelectTransport returns transport name by method, path and queryParams
+func (m *Matcher) SelectTransport(method, path, queryParams string) (transportName string) {
```

Comment:
Readability: Missleading method name it does not return transport, but transport name thus method should be called SelectTransportName